### DISCUSSION TITLE
Automatic Markers for Class Names

### DISCUSSION THREAD

**Discussion by PvtPuddles** [Nasller/CodeGlancePro#148] [Ideas]
Hello,

I'd like to start by saying this is a very useful plugin.  I've just spent the last 15 minutes playing with the `MARK: ` regex and I am very pleased.

On a similar note, I find myself writing something like the following *all. the time.*  I would love to see an option to automatically mark class definitions for easy navigation

```dart
// MARK: SomeClass
class SomeClass extends SomeOtherClass {
    // ...
}
```

I understand that this can get tricky with all the different languages out there, but I imagine creating a regex that matches `class [NAME]` would be sufficient for most cases.

**Comment by Nasller** [comment]
This may require language-specific processing

**Comment by Nasller** [comment]
try this EAP version: https://plugins.jetbrains.com/plugin/18824-codeglance-pro/edit/versions/eap

**Reply by PvtPuddles** [reply]
I am currently using `1.9.1-SNAPSHOT-2024-09-12T17-10` but I am unable to figure out how to enable the new class marker.

Am I missing a configuration option somewhere?  Do I need to add something special to the "Markers regex" (I tried adding `class ` to my RegEx but it still only seems to work in comments)?

**Reply by Nasller** [reply]
What language are you using, at the moment I only support java

**Reply by Nasller** [reply]
if you want disable ,see this options
![image](https://github.com/user-attachments/assets/fd519564-6ab4-490d-9351-35de978d59fd)

**Reply by Nasller** [reply]
kotlin support now, download lastest EAP version

**Reply by PvtPuddles** [reply]
> ..., at the moment I only support java

I had previously tried creating a java file in my non-java project just to see, but that didn't seem to work.  I've just created a test project with some sample code, and it seems to be working.  I see the marker and it looks as expected; thanks, this is super cool.

> What language are you using, ...

We use Scala for our back end and Dart for our front end.  I understand that Dart does not have a ton of support in IntelliJ but dart/flutter code tends to be quite dense so having these markers there too would be a huge help.

**Reply by Nasller** [reply]
i have support scala and dart file type, please give it some test in the lastest EAP version

**Reply by PvtPuddles** [reply]
This is incredible, thank you!

There are a couple additional cases you might want to support, for example:
- Dart
  - `interface` (beware that there is also such thing as an `interface class`)
  - `mixin` (beware that there is also such thing as an `mixin class`)
  - `extension`
- Scala
  - `trait`
  - `object`

Even without these, this package is amazing and I appreciate your work on it.  I'm going to keep playing with it and I'll let you know if I run into any issues

**Reply by Nasller** [reply]
I don't know much about dart grammar. Judging from the information collected on the Internet, mixin syntax and extension syntax can be supported, but I haven't found a good example of interface class syntax, but I think it should be highlighted, because it contains the class keyword. If not, please let me know (sorry, I didn't install the dart environment locally to test)

**Reply by Nasller** [reply]
Regarding the two grammars of scala, I will consider adding them to the highlight.

**Reply by Nasller** [reply]
> This is incredible, thank you!
> 
> There are a couple additional cases you might want to support, for example:
> 
> * Dart
>   
>   * `interface` (beware that there is also such thing as an `interface class`)
>   * `mixin` (beware that there is also such thing as an `mixin class`)
>   * `extension`
> * Scala
>   
>   * `trait`
>   * `object`
> 
> Even without these, this package is amazing and I appreciate your work on it. I'm going to keep playing with it and I'll let you know if I run into any issues

The syntaxes mentioned above have all been implemented with highlighting

**Reply by Nasller** [reply]
download the lastest EAP version
